core/src/main/java/org/apache/iceberg/avro/ValueReaders.java

```diff
@@ -739,17 +956,18 @@ protected void set(R struct, int pos, Object value) {
     }
   }
 
-  static class PositionReader implements ValueReader<Long> {
-    private long currentPosition;
-
-    PositionReader(long rowPosition) {
-      this.currentPosition = rowPosition - 1;
-    }
+  static class PositionReader implements ValueReader<Long>, SupportsRowPosition {
```

Comment:
This refactor includes changes to make `PositionReader` easier to use. Rather than needing to instantiate it inside of `setRowPositionSupplier`, it now just implements `SupportsRowPosition`. The logic for creating a position reader is now part of the object model rather than part of `StructReader`.